contracts/ensWildcardResolvers/DigidaigakuResolver.sol

```diff
@@ -0,0 +1,148 @@
+import { INameWrapper } from "../ensGuilds/GuildsResolver.sol";
+import { ReverseClaimer } from "@ensdomains/ens-contracts/contracts/reverseRegistrar/ReverseClaimer.sol";
+import { IERC721 } from "@openzeppelin/contracts/token/ERC721/IERC721.sol";
+import { IERC721Metadata } from "@openzeppelin/contracts/token/ERC721/extensions/IERC721Metadata.sol";
+import { ENS } from "@ensdomains/ens-contracts/contracts/registry/ENS.sol";
+import { Strings } from "@openzeppelin/contracts/utils/Strings.sol";
+import { Ownable } from "@openzeppelin/contracts/access/Ownable.sol";
+
+contract DigidaigakuResolver is WildcardResolverBase, ReverseClaimer, Ownable {
+    using StringParsing for bytes;
+    using Strings for string;
+    using Strings for address;
+    using Strings for uint256;
+
+    string public url;
+    IERC721 public digidaigakuContract;
+
+    constructor(
+        ENS _ensRegistry,
+        INameWrapper _ensNameWrapper,
+        address reverseRecordOwner,
+        address tokenContract,
+        string memory _url
+    ) WildcardResolverBase(_ensRegistry, _ensNameWrapper) ReverseClaimer(_ensRegistry, reverseRecordOwner) {
+        digidaigakuContract = IERC721(tokenContract);
+        url = _url;
+        return;
+    }
+
+    function setUrl(string memory newUrl) public onlyOwner {
+        url = newUrl;
+    }
+
+    function _resolveWildcardEthAddr(
+        bytes calldata childUtf8Encoded,
+        bytes calldata parentDnsEncoded
+    ) internal view virtual override returns (address) {
+        // Extract tokenId from child name
+        (bool valid, uint256 tokenId) = childUtf8Encoded.parseUint256();
+        // No token, try resolving using name
+        if (!valid) {
+            string[] memory urls = new string[](1);
+            urls[0] = url;
+
+            revert OffchainLookup(
+                address(this),
+                urls,
+                childUtf8Encoded,
+                this.resolveByNameCallback.selector,
+                parentDnsEncoded
+            );
+        }
+
+        return resolveOwnerAddress(tokenId);
+    }
+
+    function resolveByNameCallback(
+        bytes calldata response,
+        bytes calldata extraData // solhint-disable-line no-unused-vars
+    ) public view returns (bytes memory) {
+        // Get tokenId from offchain response
+        uint256 tokenId = abi.decode(response, (uint256));
+        return abi.encode(resolveOwnerAddress(tokenId));
+    }
+
+    function resolveOwnerAddress(uint256 tokenId) internal view returns (address) {
+        // Lookup token owner
+        address tokenOwner;
+        try digidaigakuContract.ownerOf(tokenId) returns (address _tokenOwner) {
+            tokenOwner = _tokenOwner;
+        } catch {
+            tokenOwner = address(0);
+        }
+        return tokenOwner;
+    }
+
+    function _resolveWildcardTextRecord(
+        bytes calldata childUtf8Encoded,
+        bytes calldata parentDnsEncoded, // solhint-disable-line no-unused-vars
```

Comment:
Prob want to make this just empty bytes then so it's less confusing for the reader why we're including unused extraData